File: Fr8Infrastructure.NET/Data/DataTransferObjects/AuthenticationTokenDTO.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Fr8.Infrastructure.Data.DataTransferObjects
{
    public class AuthenticationTokenTerminalDTO
    {
        [JsonProperty("id")]
        public Guid Id { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("version")]
        public string Version { get; set; }

        [JsonProperty("label")]
        public string Label { get; set; }

        [JsonProperty("authenticationType")]
        public int AuthenticationType { get; set; }
        
        [JsonProperty("authTokens")]
        public List<AuthenticationTokenDTO> AuthTokens { get; set; }
    }

    public class AuthenticationTokenDTO
    {
        [JsonProperty("id")]
        public Guid Id { get; set; }

        [JsonProperty("externalAccountName")]
        public string ExternalAccountName { get; set; }

        [JsonProperty("isMain")]
        public bool IsMain { get; set; }

        [JsonProperty("isSelected")]
        public bool IsSelected { get; set; }
    }

    public class AuthenticationTokenGrantDTO
    {
        [JsonProperty("actionId")]
        public Guid ActivityId { get; set; }

        [JsonProperty("authTokenId")]
        public Guid AuthTokenId { get; set; }

        [JsonProperty("isMain")]
        public bool IsMain { get; set; }
    }
}

File: Hub/HubWeb/Scripts/app/model/AuthenticationTokenDTO.ts
module dockyard.model {

    export class AuthenticationTokenTerminalDTO {
        name: string;
        label: string;
        version: string;
        authTokens: Array<AuthenticationTokenDTO>;
        authenticationType: number;
        selectedAuthTokenId: number;

        constructor(
            name: string,
            authTokens: Array<AuthenticationTokenDTO>,
            authenticationType: number,
            selectedAuthTokenId: number) {

            this.name = name;
            this.authTokens = authTokens;
            this.authenticationType = authenticationType;
            this.selectedAuthTokenId = selectedAuthTokenId;
        }
    }

    export class AuthenticationTokenDTO {
        id: number;
        externalAccountName: string;
        isMain: boolean;
        isSelected: boolean;

        constructor(
            id: number,
            externalAccountName: string,
            isMain: boolean,
            isSelected: boolean
        ) {
            this.id = id;
            this.externalAccountName = externalAccountName;
            this.isMain = isMain;
            this.isSelected = isSelected;
        }
    }
}